### ISSUE TITLE
prost_types::Value to wkt::Value conversion panics on NaN or infinity

### ISSUE THREAD

**Issue by coryan** [googleapis/google-cloud-rust#6707]
Unconfirmed.  This is the result of a search with Claude, may be a false positive.

In `src/gax-internal/src/prost.rs`, converting `prost_types::Value` to `wkt::Value` handles `Kind::NumberValue(v)`:

https://github.com/googleapis/google-cloud-rust/blob/f02b42c2ac5720334cd2b5335824bd4166264527/src/gax-internal/src/prost.rs#L169-L173

```rust
Kind::NumberValue(v) => {
    let number =
        serde_json::Number::from_f64(v).expect("JSON numbers cannot be NaN");
    serde_json::Value::Number(number)
}
```

While JSON does not permit `NaN` or `Infinity`, protobuf binary encoding for `double` fields can carry `NaN`, `+Infinity`, and `-Infinity`.

When a gRPC response containing `NaN` or `Infinity` in a `google.protobuf.Value` field is received, `.expect()` panics and crashes the client. The function signature already returns `Result<wkt::Value>`, so an error should be returned instead.
